envoy/service/accesslog/v2/als.proto

```diff
@@ -0,0 +1,100 @@
+
+package envoy.service.accesslog.v2;
+option go_package = "accesslog";
+
+import "envoy/api/v2/base.proto";
+import "envoy/api/v2/grpc_service.proto";
+import "envoy/api/v2/filter/accesslog/accesslog.proto";
+
+import "validate/validate.proto";
+
+// Service for streaming access logs from Envoy to an access log server.
+service AccessLogService {
+  // Envoy will connect and send StreamAccessLogsMessage messages forever. It does not expect any
+  // response to be sent as nothing would be done in the case of failure. The server should
+  // disconnect if it expects Envoy to reconnect. In the future we may decide to add a different
+  // API for "critical" access logs in which Envoy will buffer access logs for some period of time
+  // until it gets an ACK so it could then retry. This API is designed for high throughput with the
+  // expectation that it might be lossy.
+  rpc StreamAccessLogs(stream StreamAccessLogsMessage) returns (StreamAccessLogsResponse) {
+  }
+}
+
+// Empty response for the StreamAccessLogs API. Will never be sent. See below.
+// [#not-implemented-hide:] Not configuration. TBD how to doc proto APIs.
+message StreamAccessLogsResponse {
+}
+
+// [#proto-status: experimental]
+// [#not-implemented-hide:] Not configuration. TBD how to doc proto APIs.
+// Stream message for the StreamAccessLogs API. Envoy will open a stream to the server and stream
+// access logs without ever expecting a response.
+message StreamAccessLogsMessage {
+  message Identifier {
+    // The node sending the access log messages over the stream.
+    envoy.api.v2.Node node = 1 [(validate.rules).message.required = true];
+
+    // The friendly name of the log configured in AccessLogServiceConfig.
+    string log_name = 2 [(validate.rules).string.min_bytes = 1];
+  }
+
+  // Identifier data that will only be sent in the first message on the stream. This is effectively
+  // structured metadata and is a performance optimization.
+  Identifier identifier = 1;
+
+  // Wrapper for batches of HTTP access log entries.
+  message HTTPAccessLogEntries {
+    repeated envoy.api.v2.filter.accesslog.HTTPAccessLogEntry log_entry = 1
+        [(validate.rules).repeated .min_items = 1];
+  }
+
+  // Wrapper for batches of TCP access log entries.
+  message TCPAccessLogEntries {
+    repeated envoy.api.v2.filter.accesslog.TCPAccessLogEntry log_entry = 1
+        [(validate.rules).repeated .min_items = 1];
+  }
+
+  // Batches of log entries of a single type. Generally speaking, a given stream should only
+  // ever incude one type of log entry.
+  oneof log_entries {
+    option (validate.required) = true;
+
+    HTTPAccessLogEntries http_logs = 2;
+    TCPAccessLogEntries tcp_logs = 3;
+  }
+}
+
+// Configuration for the built-in *envoy.tcp_grpc_access_log* type. This configuration will
+// populate *StreamAccessLogsMessage.tcp_logs*.
+// [#not-implemented-hide:]
+// [#comment:TODO(mattklein123): Block type in non-tcp proxy cases?]
+message TcpGrpcAccessLogConfig {
```

Comment:
What's the rationale for locating the configuration (which is consumed by Envoy) in the same place as the service definition (which is the used by both external entities and Envoy)? Maybe we should have `envoy.config`, where both bootstrap and other non-filter config can live.

Reply:
My rationale was to preserve the versioning from the service. A service config for v2 seems to belong to v2 service definition. There's now `envoy.bootstrap.v2` solely for bootstrap config, which might also be appropriate place. WDYT?

Reply:
I think the service config is something that Envoy understands which is somewhat independent of the service interface as understood by Envoy and management server. Logically you could version one without the other. But, if it's simpler you can stick with what is there, I just felt that there is opportunity for further separation of concerns here by having a `envoy.config` hierarchy.

Reply:
@mattklein123 @wora any preference here. I think we have three choices:
1. keep service config next to service definition (in the same version directory)
2. move service config to `envoy/bootstrap/v2`, the startup config for Envoy
3. create new hierarchy `envoy/config/accesslog/v2` for service configs

Reply:
I have a small preference for (1) as I think it's easier to understand, but I don't care that much.

Reply:
I am ok with any of the choice. I've prepared https://github.com/envoyproxy/envoy/pull/2430 but hitting some linker issues at the moment.

Reply:
@wora do you have any thoughts on whether the proxy configuration related to a service and the service definition itself should be colocated in a package or placed in different namespaces?

Reply:
Related to this: we could possibly move non-Envoy service configs, too, out of core, for zipkin, lightstep, statsd.